src/SignalR/common/Protocols.MessagePack/src/MessagePackHubProtocolOptions.cs

```diff
@@ -9,24 +9,24 @@ namespace Microsoft.AspNetCore.SignalR
 {
     public class MessagePackHubProtocolOptions
     {
-        private IList<IFormatterResolver> _formatterResolvers;
+        private MessagePackSerializerOptions _messagePackSerializerOptions;
 
-        public IList<IFormatterResolver> FormatterResolvers
+        public MessagePackSerializerOptions SerializerOptions
```

Comment:
Lets add xml comments.

* Recommend setting `.WithSecurity(MessagePackSecurity.UntrustedData)` if they override the options
* If you want to modify to the default options you need to assign the options back to the `SerializerOptions` after modifications:
```c#
SerializerOptions = SerializerOptions.WithResolver(new CustomResolver());
```

Reply:
quick question, as the API has been approuved, and thinking about the summary, i have a terrible idea....
(i have no idea if it's possible) but does adding this makes sense:
(pseudo code, i have no idea if it's something good or bad since it's intrusive and also how properly speak english and being both nice and clear to the customer)
```
if(env.IsDevelopment() && options.serializerOptions.MessagePackSecurity != MessagePackSecurity.UntrustedData)
{
  if(options.EnableWarningForUntrustedData)
  {
    logger.LogWarning("MessagePackSerializerOptions have been overriden with a ... security settings ...blablabla if intended, you can turn this warning off by settings EnableWarningForUntrustedData to false")

   could even points to a github issue explaining that API

  }
}
```

Reply:
The `options.serializerOptions.MessagePackSecurity != MessagePackSecurity.UntrustedData` would need some massaging because `UntrustedData` is a copy-on-write object so if they've 'enhanced' it at all, you wouldn't want that to reactivate the failure. You'd want to look *into* the object to see what settings it has to determine whether it's a secure instance. But let's first wait to see what the ASP.NET Core folks think of your overall idea.

Reply:
as said it just something that crossed my mind while trying to find an idea for the summary (TBH I'm pretty bad at it)
Also i have no idea if it's even possible

Reply:
not sure about it:
![image](https://user-images.githubusercontent.com/2266487/78143372-78aad280-742e-11ea-81fe-b7f10a072f1a.png)

Reply:
@anurse to help with word smithing

Reply:
Current rendering
![image](https://user-images.githubusercontent.com/2266487/78282114-0adcd480-751c-11ea-8db5-57611de4a841.png)

Reply:
I'm happy with that 😄 

I'll give folks till the end of the day to comment, otherwise I'll merge

Reply:
![9F4F2CC0-8025-4A38-8E41-B157D003F475](https://user-images.githubusercontent.com/2266487/78289480-8e012900-7522-11ea-8d94-e5dbb2499dd5.gif)

Reply:
welllll
I should remove the `.` since the `<code>` seems to create a paragraph right 
@anurse  ?

![image](https://user-images.githubusercontent.com/2266487/78290507-1cc27580-7524-11ea-8c66-d0b9ad250064.png)

Reply:
(pushed a revert commit to this) :

![image](https://user-images.githubusercontent.com/2266487/78290887-bbe76d00-7524-11ea-9b8e-c2fa2a8f3c4d.png)

Reply:
Good point. Yep, drop the `.` after the `<code>`.

Reply:
(done, do i mark this thread as resolved if summary if ok for you ?)